### PR TITLE
refactor: remove cniTypesCurr.Result dependency in InterfaceInfo

### PR DESCRIPTION
**Pull request by jaer-tsun**
**Reason for Change**:

removing cniTypesCurr.Result dependency in InterfaceInfo

adding scope link for swiftv2 gateway route

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by nddq — APPROVED** [review]

File: cni/network/network.go [hunk 1]
```diff
@@ -1362,18 +1343,61 @@ func convertNnsToCniResult(
...
-				ipConfig := &cniTypesCurr.IPConfig{
-					Address:   address,
-					Gateway:   gateway,
-					Interface: &intIndex,
-				}
 
-				resultIpconfigs = append(resultIpconfigs, ipConfig)
+				ipConfigs = append(ipConfigs, &network.IPConfig{
+					Address: address,
+					Gateway: gateway,
```
Comment (vipul-21) [inline cni/network/network.go:1349]: we are not interested in interface index ? Was there a use case related to it ?

**Review by vipul-21 — APPROVED** [review]
can we run cni load test pipeline as well for this PR ?

**Comment by jaer-tsun** [comment]
> can we run cni load test pipeline as well for this PR ?

load tests passed https://msazure.visualstudio.com/One/_build/results?buildId=82852995&view=results

**Pull request merged**